packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -49,9 +49,10 @@ abstract class ScrollView extends StatelessWidget {
     this.reverse: false,
     this.controller,
     bool primary,
-    this.physics,
+    ScrollPhysics physics,
     this.shrinkWrap: false,
   }) : primary = primary ?? controller == null && scrollDirection == Axis.vertical,
+       physics = physics ?? (primary == true || (primary == null && controller == null && scrollDirection == Axis.vertical) ? const AlwaysScrollableScrollPhysics() : null),
```

Comment:
On the left hand side of an initializer, only the fields are in scope. On the right hand side, only the constructor arguments are in scope. *shrug*

(In particular, this is why the `physics=` here has to duplicate the logic that set `primary`, rather than being able to just reuse it.)